### PR TITLE
[AAP-70261] Fix role user assignment

### PR DESCRIPTION
**Pull request by nickbhasin**
## Description

- What is being changed?
The hardcoded role name check is being replaced with a dynamic API endpoint mapping derived directly from the role definition's content_type attribute.
- Why is this change needed?
The module previously failed to assign custom roles that didn't start with specific prefixes.
- How does this change address the issue?
It correctly extracts the base object type from the content_type string (e.g., mapping shared.organization to organizations) to ensure the API request routes to the correct destination regardless of the role's name.

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Test update
- [ ] Refactoring (no functional changes)
- [ ] Development environment change
- [ ] Configuration change

## Self-Review Checklist

- [x] I have performed a self-review of my code
- [ ] I have added relevant comments to complex code sections
- [ ] I have updated documentation where needed
- [ ] I have considered the security impact of these changes
- [ ] I have considered performance implications
- [ ] I have thought about error handling and edge cases
- [ ] I have tested the changes in my local environment

### Steps to Test
1. 
2. 
3. 

### Required Actions

- [ ] Requires documentation updates
  
- [ ] Requires downstream repository changes
  
- [ ] Requires infrastructure/deployment changes
  
- [ ] Requires coordination with other teams
  
- [ ] Blocked by PR/MR: #XXX

### CODE REVIEW

File: tests/integration/targets/role_user_assignments_test/tasks/main.yml [hunk 1]
```diff
@@ -223,6 +224,61 @@
         that:
           - team2_admin_role_assignment_check is not changed
 
+    - name: Create custom role for role_user_assignment integration test
+      ansible.platform.role_definition:
+        name: "{{ custom_role_name }}"
+        description: "Custom role for role_user_assignment content_type resolution test"
+        content_type: "shared.organization"
```
Comment (rohitthakur2590) [inline tests/integration/targets/role_user_assignments_test/tasks/main.yml:232]: can we also cover test for other added content types

**Review by rohitthakur2590 — COMMENTED** [review]
lets also fix this in 2.7 .

File: tests/integration/targets/role_user_assignments_test/tasks/main.yml (hunk 1 above)
Comment (rohitthakur2590) [reply tests/integration/targets/role_user_assignments_test/tasks/main.yml:232]: lets also update documentation with examples

**Comment by nickbhasin** [comment]
> lets also fix this in 2.7 .
its working in 2.7

File: tests/integration/targets/role_user_assignments_test/tasks/main.yml (hunk 1 above)
Comment (nickbhasin) [reply tests/integration/targets/role_user_assignments_test/tasks/main.yml:232]: added/updated

**Review by rohitthakur2590 — APPROVED** [review]
